### DISCUSSION TITLE
Smooth Google Icons with 7789 TFT

### DISCUSSION THREAD

**Discussion by MYCAMEL222** [Bodmer/TFT_eSPI#2315] [Q&A - General] [answered]
I've got the font conversion to smooth fonts down pat using Bodmer's method. 

I'm trying to figure out how to take Google Icons (https://fonts.google.com/icons) and turn them into 48x48 smooth graphics.

Currently, I use one of the online converters to convert from PNG to hex and store that as const unsigned char array, then use drawbmp.

They are never "crisp". 

Is drawXbmp a better solution? or is there a solution. 

Is displaying a 48x48 JPEG a better way?

I'm all ears......Thanks

**Comment by Bodmer** [comment]
If your processor has sufficient RAM then you can render a PNG direct to the screen. Simple icons typically end up as very smallfiles that way. See example.

Another option is to find a font with suitable icons.

**Comment by Bodmer** [comment, marked as answer]
I have added an example that may be of interest, it renders PNG images that have a transparent area. Example here:
https://github.com/Bodmer/TFT_eSPI/tree/master/examples/PNG%20Images/Flash_transparent_PNG

**Reply by MYCAMEL222** [reply]
I decided, why use monochrome icons when I'm using a full color TFT....?

I designed/copied some icons from Canva and downloaded them as 48x48 jpegs with the appropriate color as a background. The JPG  did not meet my quality hopes.

I redid all of them as PNG, again with the correct background colors and that are spot on . No need to even try the transparent background (thanks for posting the demo).

And thank you for a great library.!!

Now I need to figure out how to print the PNG to a sprite, but that is for another day...


![20230113_151234](https://user-images.githubusercontent.com/85456085/212420296-ea5d99b1-efdd-42da-850b-64a924898ef6.jpg)
